Meta/http-test-server.py

```diff
@@ -0,0 +1,334 @@
+        elif self.path == "/ping":
+            self.send_response(200)
+            self.send_header("Content-Type", "text/plain")
+            self.end_headers()
+            self.wfile.write(b"pong")
+        elif self.path.startswith("/echo/"):
+            self.handle_echo()
+        elif self.path.startswith("/static/"):
+            # Remove '/static/' prefix and use built-in method
+            self.path = self.path[7:]
+            return super().do_GET()
+        else:
+            self.send_error(404, "Not Found")
+
+    def do_POST(self):
+        if self.path == "/create":
+            content_length = int(self.headers["Content-Length"])
+            post_data = self.rfile.read(content_length)
+            response_def = json.loads(post_data.decode("utf-8"))
+
+            response_id = str(len(echo_store) + 1)
+            echo_store[response_id] = response_def
+
+            self.send_response(201)
+            self.send_header("Access-Control-Allow-Origin", "*")
+            self.send_header("Content-Type", "application/json")
+            self.end_headers()
+            self.wfile.write(json.dumps({"id": response_id}).encode("utf-8"))
+        elif self.path.startswith("/static/"):
+            self.send_error(405, "Method Not Allowed")
+        else:
+            self.handle_echo()
+
+    def do_OPTIONS(self):
+        if self.path.startswith("/create"):
+            self.send_response(204)
+            self.send_header("Access-Control-Allow-Origin", "*")
+            self.send_header("Access-Control-Allow-Methods", "*")
+            self.send_header("Access-Control-Allow-Headers", "*")
+            self.end_headers()
+        else:
+            self.do_other()
+
+    def do_PUT(self):
+        self.do_other()
+
+    def do_HEAD(self):
+        self.do_other()
+
+    def do_DELETE(self):
+        self.do_other()
+
+    def handle_echo(self):
+        parsed_path = urlparse(self.path)
+        response_id = parsed_path.path.split("/")[-1]
+
+        if response_id in echo_store:
+            response_def = echo_store[response_id]
+
+            if "delay" in response_def:
+                time.sleep(response_def["delay"])
+
+            # Send the status code without any default headers
+            self.send_response_only(response_def.get("status", 200))
+
+            # Set only the headers defined in the echo definition
+            for header, value in response_def.get("headers", {}).items():
+                self.send_header(header, value)
+            self.end_headers()
+
+            self.wfile.write(response_def.get("body", "").encode("utf-8"))
+        else:
+            self.send_error(404, "Echo response not found")
+
+    def do_other(self):
+        if self.path.startswith("/echo/"):
+            self.handle_echo()
+        elif self.path.startswith("/static/"):
+            self.send_error(405, "Method Not Allowed")
+        else:
+            self.send_error(404, "Not Found")
+
+    def get_root_html(self):
```

Comment:
I'm not sure if we should have a root page containing usage instructions. Ideally, the test server is very barebones and contains one hardcoded path to setup test responses. Documentation should live in a dedicated markdown file IMO.

Reply:
I can move the docs out to a .md file if the HTML version is adding undue complexity/distraction. 👍🏻

Reply:
I don't think it's a bad idea, just think it's easier to reason about when we have a test HTTP server dedicated to one task.